com.github.lruzicka.Needly.yaml

```diff
@@ -0,0 +1,77 @@
+# The YAML manifest for the Needle Flatpak
+#
+app-id: com.github.lruzicka.Needly
+runtime: org.freedesktop.Platform
+runtime-version: '21.08'
+sdk: org.freedesktop.Sdk
+command: needly
+finish-args: 
+   - --share=ipc # For X11
+   - --socket=x11 # Enable communication over X11
+   - --filesystem=home # Allow read-write access to home
+   - --filesystem=xdg-run/gvfs # Access gvfs mounts
+   - --filesystem=xdg-run/libvirt # Access to qemu:///session
+   - --filesystem=/run/libvirt # Access to qemu:///system
+
+# Installing the application and the dependencies
+#
+modules:
+  # This will install the Tkinter support for Python as
+  # Tkinter is not part of the runtime.
+  - python3-tkinter.yaml
+
+  # This is a dependency for libvirt.
+  - name: libtirpc
+    config-opts:
+      - --disable-gssapi
+      - --libdir=/app/lib
+    sources:
+      - type: archive
+        url: https://downloads.sourceforge.net/sourceforge/libtirpc/libtirpc-1.3.2.tar.bz2
+        sha256: e24eb88b8ce7db3b7ca6eb80115dd1284abc5ec32a8deccfed2224fc2532b9fd
+
+  # This is another dependency for libvirt which is used to
+  # create the necessary documentation.
+  - name: docutils
+    buildsystem: simple
+    build-commands: 
+      - pip3 install --prefix=/app docutils-0.16-py2.py3-none-any.whl
+    sources: 
+      - type: file
+        url: https://files.pythonhosted.org/packages/81/44/8a15e45ffa96e6cf82956dd8d7af9e666357e16b0d93b253903475ee947f/docutils-0.16-py2.py3-none-any.whl
+        sha256: 0c5b78adfbf7762415433f5515cd5c9e762339e23369dbe8000d84a4bf4ab3af
+
+  # Another libvirt dependency
+  - name: rpcsvc-proto
+    buildsystem: autotools
+    sources:
+    - type: archive
+      url: https://github.com/thkukuk/rpcsvc-proto/archive/v1.4.1.tar.gz
+      sha256: 750f7e57b81407a25b707867e90d7ee80aeb53bf515b114fc218f3c78dc9a6e8
+
+  # Libvirt
+  - name: libvirt
+    buildsystem: meson
+    build-options:
+    - cflags: -I/app/include/tirpc -ltirpc
+    config-opts:
+      - --localstatedir=/var
+    sources:
+    - type: archive
+      url: https://libvirt.org/sources/libvirt-8.0.0.tar.xz
+      sha256: 51e6e8ff04bafe96d7e314b213dcd41fb1163d9b4f0f75cdab01e663728f4cf6
+
+  # Python Libvirt API
+  - name: libvirt-python
+    buildsystem: simple
+    build-commands: 
+      - pip3 install --prefix=/app libvirt-python-8.1.0.tar.gz
+    sources: 
+      - type: file
+        url: https://files.pythonhosted.org/packages/2a/74/919462bed158ccc2a8061f31d82c157740c2e597ff50490526336e8aa688/libvirt-python-8.1.0.tar.gz
+        sha256: a21ecfab6d29ac1bdd1bfd4aa3ef58447f9f70919aefecd03774613f65914e43
+
+  # The Needly application
+  - python3-needly.json
```

Comment:
the main module for the application should be in the main manifest. alternative are difficult to maintain.

Reply:
> the main module for the application should be in the main manifest. alternative are difficult to maintain.

I put the Needly module into the main manifest.